lib/ansible/galaxy/api.py

```diff
@@ -796,6 +836,60 @@ def get_collection_version_metadata(self, namespace, name, version):
                                          data['download_url'], data['artifact']['sha256'],
                                          data['metadata']['dependencies'], data['href'], signatures)
 
+    @g_connect(['v2', 'v3'])
+    def reset_cache(self, versions_url, namespace, name, cache_key):
+        # We should only rely on the cache if the collection has not changed. This may slow things down but it ensures
+        # we are not waiting a day before finding any new collections that have been published.
+        if self._cache:
+            server_cache = self._cache.setdefault(get_cache_id(versions_url), {})
+            modified_cache = server_cache.setdefault('modified', {})
+
+            try:
+                modified_date = self.get_collection_metadata(namespace, name).modified_str
+            except GalaxyError as err:
+                if err.http_code != 404:
+                    raise
+                # No collection found, return an empty list to keep things consistent with the various APIs
+                return []
+
+            cached_modified_date = modified_cache.get('%s.%s' % (namespace, name), None)
+            if cached_modified_date != modified_date:
+                modified_cache['%s.%s' % (namespace, name)] = modified_date
+                if cache_key in server_cache:
+                    del server_cache[cache_key]
+
+                self._set_cache()
+
+    @g_connect(['v2', 'v3'])
+    def set_default_collection_page_size(self, api_path, namespace, name):
+        versions_url = _urljoin(self.api_server, api_path, 'collections', namespace, name, 'versions')
+        cache_key = urlparse(versions_url).path
+        self.reset_cache(versions_url, namespace, name, cache_key)
+
+        error_context_msg = 'Error when getting available collection versions for %s.%s from %s (%s).' \
+                            % (namespace, name, self.name, self.api_server)
+
+        try:
+            data = self._call_galaxy(versions_url, error_context_msg=error_context_msg, cache=True, cache_key=cache_key)
+        except GalaxyError as err:
+            if err.http_code != 404:
+                raise
+            # v3 doesn't raise a 404 so we need to mimick the empty response from APIs that do.
+            return
+
+        if 'data' in data:
+            # v3 automation-hub is the only known API that uses `data`
+            # since v3 pulp_ansible does not, we cannot rely on version
+            # to indicate which key to use
+            results_key = 'data'
+        else:
+            results_key = 'results'
+
+        if 'count' in data and data['count'] < len(data[results_key]) * COLLECTION_PAGE_MULTIPLIER:
```

Comment:
Would this be more elegant?
```suggestion
        if data.get('count', float('inf')) < len(data[results_key]) * COLLECTION_PAGE_MULTIPLIER:
```